lib/presentation/profile/medical_data/medical_data_page.dart

```diff
@@ -1,41 +1,149 @@
+import 'package:myafyahub/application/redux/view_models/medical_data_view_model.dart';
+import 'package:myafyahub/domain/core/entities/medical_data/medical_data_details.dart';
 // Project imports:
 import 'package:myafyahub/domain/core/value_objects/app_strings.dart';
 import 'package:myafyahub/domain/core/value_objects/app_widget_keys.dart';
 import 'package:myafyahub/domain/core/value_objects/asset_strings.dart';
 import 'package:myafyahub/presentation/core/theme/theme.dart';
 import 'package:myafyahub/presentation/core/widgets/app_bar/custom_app_bar.dart';
+import 'package:myafyahub/presentation/profile/medical_data/medical_data_item.dart';
+import 'package:myafyahub/presentation/profile/medical_data/medical_data_item_title.dart';
 
-class MedicalDataPage extends StatelessWidget {
+class MedicalDataPage extends StatefulWidget {
   const MedicalDataPage({Key? key}) : super(key: key);
 
+  @override
+  State<MedicalDataPage> createState() => _MedicalDataPageState();
+}
+
+class _MedicalDataPageState extends State<MedicalDataPage> {
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    StoreProvider.dispatch(
+      context,
+      FetchMedicalDataAction(
+        httpClient: AppWrapperBase.of(context)!.graphQLClient,
+      ),
+    );
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: const CustomAppBar(title: medicalDataPageTitle),
       backgroundColor: Colors.white,
       body: SafeArea(
-        child: Padding(
-          padding: const EdgeInsets.all(20.0),
-          child: GenericErrorWidget(
-            headerIconSvgUrl: zeroMedicalDataSvgPath,
-            actionKey: helpNoDataWidgetKey,
-            actionText: okThanksText,
-            recoverCallback: () {
-              Navigator.of(context).pop();
-            },
-            messageTitle: noMedicalDataString,
-            messageBody: <TextSpan>[
-              TextSpan(
-                text: noMedicalDataBodyString,
-                style: normalSize16Text(
-                  AppColors.greyTextColor,
-                ),
-              ),
-            ],
-          ),
+        child: StoreConnector<AppState, MedicalDataViewModel>(
+          converter: (Store<AppState> store) =>
+              MedicalDataViewModel.fromStore(store),
+          builder: (BuildContext context, MedicalDataViewModel vm) {
+            if (vm.wait!.isWaitingFor(fetchMedicalDataFlag)) {
+              return Container(
+                height: 300,
+                padding: const EdgeInsets.all(20),
+                child: const PlatformLoader(),
+              );
+            } else {
+              final MedicalDataDetails? alergies = vm.medicalData?.allergies;
+              final MedicalDataDetails? viralLoad = vm.medicalData?.viralLoad;
+              final MedicalDataDetails? bmi = vm.medicalData?.bmi;
+              final MedicalDataDetails? cd4Count = vm.medicalData?.cd4Count;
```

Comment:
You can have them default to `MedicalDataDetails.initial()` so that you won't have to check for their `nullness` in L61 below
```suggestion
              final MedicalDataDetails alergies = vm.medicalData?.allergies ?? MedicalDataDetails.initial();
              final MedicalDataDetails viralLoad = vm.medicalData?.viralLoad ?? MedicalDataDetails.initial();
              final MedicalDataDetails bmi = vm.medicalData?.bmi ?? MedicalDataDetails.initial();
              final MedicalDataDetails cd4Count = vm.medicalData?.cd4Count ?? MedicalDataDetails.initial();
```